components/webhook-mgt/org.wso2.carbon.identity.subscription.management/src/main/java/org/wso2/carbon/identity/subscription/management/internal/service/impl/SubscriptionManagementServiceImpl.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2025, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.subscription.management.internal.service.impl;
+
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.subscription.management.api.constant.ErrorMessage;
+import org.wso2.carbon.identity.subscription.management.api.exception.SubscriptionManagementException;
+import org.wso2.carbon.identity.subscription.management.api.model.ChannelSubscriptionRequest;
+import org.wso2.carbon.identity.subscription.management.api.model.Subscription;
+import org.wso2.carbon.identity.subscription.management.api.service.EventSubscriber;
+import org.wso2.carbon.identity.subscription.management.api.service.SubscriptionManagementService;
+import org.wso2.carbon.identity.subscription.management.internal.component.SubscriptionManagementComponentServiceHolder;
+import org.wso2.carbon.identity.subscription.management.internal.util.SubscriptionManagementExceptionHandler;
+
+import java.util.List;
+
+/**
+ * Implementation of the SubscriptionManagementService interface.
+ * This class provides implementation for subscription management operations.
+ */
+public class SubscriptionManagementServiceImpl implements SubscriptionManagementService {
+
+    private static final SubscriptionManagementServiceImpl subscriptionManagementServiceImpl =
+            new SubscriptionManagementServiceImpl();
+
+    /**
+     * Singleton instance of SubscriptionManagementServiceImpl.
+     *
+     * @return Singleton instance.
+     */
+    public static SubscriptionManagementServiceImpl getInstance() {
+
+        return subscriptionManagementServiceImpl;
+    }
+
+    @Override
+    public List<Subscription> subscribe(ChannelSubscriptionRequest subscriptionRequest, String adaptor,
+                                        String tenantDomain)
+            throws SubscriptionManagementException {
+
+        EventSubscriber subscriber = retrieveAdaptorManager(adaptor);
+        return subscriber.subscribe(subscriptionRequest, IdentityTenantUtil.getTenantId(tenantDomain));
```

Comment:
Are we passing the tenantId for this ? It should be tenantDomain. tenantId should be resolved at data layer only we should not use it at service layer

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/6915/commits/4093d26554a124dc965d55155ecb5d38eec59fb8